url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/compiler/inst/XInstConst.java

```diff
@@ -0,0 +1,174 @@
+package xscript.compiler.inst;
+
+import xscript.XOpcode;
+import xscript.compiler.XDataOutput;
+
+public class XInstConst extends XInst {
+
+	private Object _const;
+	
+	public XInstConst(int line, Object _const) {
+		super(line, getOP(_const));
+		this._const = _const;
+	}
+
+	private static XOpcode getOP(Object _const){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 36. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.